experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeContext.ts

```diff
@@ -0,0 +1,147 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils";
+import {
+	IEditableForest,
+	moveToDetachedField,
+	ForestEvents,
+	FieldStoredSchema,
+	FieldKey,
+} from "../../core";
+import { ISubscribable } from "../../events";
+import { DefaultEditBuilder } from "../default-field-kinds";
+import { NodeKeyManager } from "../node-key";
+import { FieldGenerator } from "../contextuallyTyped";
+import { TypedSchemaCollection } from "../typed-schema";
+import { disposeSymbol } from "../../util";
+import { UntypedField } from "./editableTreeTypes";
+import { makeField } from "./lazyField";
+import { LazyEntity, prepareForEditSymbol } from "./lazyEntity";
+
+/**
+ * A common context of a "forest" of EditableTrees.
+ * It handles group operations like transforming cursors into anchors for edits.
+ * @alpha
+ */
+export interface TreeContext extends ISubscribable<ForestEvents> {
+	/**
+	 * Gets the root field of the tree.
+	 */
+	get root(): UntypedField;
+
+	/**
+	 * Schema used within this context.
+	 * All data must conform to these schema.
+	 */
+	readonly schema: TypedSchemaCollection;
+}
+
+/**
+ * Implementation of `EditableTreeContext`.
+ *
+ * An editor is required to edit the EditableTrees.
+ */
+export class Context implements TreeContext {
+	public readonly withCursors: Set<LazyEntity> = new Set();
+	public readonly withAnchors: Set<LazyEntity> = new Set();
+
+	private readonly eventUnregister: (() => void)[];
+
+	/**
+	 * @param forest - the Forest
```

Comment:
Since we're using inline properties in the constructor anyways, it would probably be better to make move these param comments down as individual comments on each property.

Reply:
If you do that, you lose the docs for them when calling the constructor, and they only show up when accessing the field. At least thats what happened when I just tried it.

Reply:
Really? I could have sworn that worked. I believe it works on the generated docs side of things (though now I'm questioning everything and will have to double check).

Reply:
I just tested intellisense, which is all I use

Reply:
Interesting. Both options appear to "work" for me. And in fact, if I add comments in both places, intellisense concatenates them.

What's interesting, though, is that if the property is documented using `@inheritDoc`, the inheritance is correctly resolved if the property in question is a standard property (rather than inline), but if it's inline intellisense prints "{@inheritDoc ...}.